#VRML_SIM R2022a utf8
WorldInfo {
  basicTimeStep 8
}
Viewpoint {
  orientation 0.3203986202036012 -0.20533721688998613 -0.9247601589231141 4.211594816979355
  position 0.6018089757904347 -1.3362480692248777 1.113430275557926
}
Background {
  skyColor [
    0.231373 0.34902 0.34902
  ]
}
DirectionalLight {
  direction 0 -2 -1
  intensity 2
}
Robot {
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.07 0 0.04
      }
      device [
        RotationalMotor {
          name "headMotor"
        }
      ]
      endPoint Solid {
        translation -0.07 0 0.04
        children [
          DistanceSensor {
            translation -0.01 0.015 0
            rotation 0 0 -1 -2.8797953071795863
            children [
              Shape {
                appearance GlossyPaint {
                  baseColor 0 0 1
                }
                geometry Sphere {
                  radius 0.01
                }
              }
            ]
            name "distance sensor 2"
            lookupTable [
              0 1 0
              0.5 0 0
            ]
          }
          DistanceSensor {
            translation -0.01 -0.015 0
            rotation 0 0 1 -2.8797953071795863
            children [
              Shape {
                appearance GlossyPaint {
                  baseColor 0 0 1
                }
                geometry Sphere {
                  radius 0.01
                }
              }
            ]
            name "distance sensor 1"
            lookupTable [
              0 1 0
              0.5 0 0
            ]
          }
          Shape {
            appearance MattePaint {
              baseColor 0.333333 0.333333 0.498039
            }
            geometry DEF headGeom Box {
              size 0.02 0.05 0.02
            }
          }
        ]
        boundingObject USE headGeom
        physics Physics {
          density 1
        }
      }
    }
    LED {
      translation -0.04 0.03 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "neuron 1"
      color []
      gradual TRUE
    }
    LED {
      translation -0.04 -0.01 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "neuron 2"
      color []
      gradual TRUE
    }
    LED {
      translation 0 0.03 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "neuron 3"
      color []
      gradual TRUE
    }
    LED {
      translation 0 -0.01 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "neuron 4"
      color []
      gradual TRUE
    }
    LED {
      translation 0.2 0.2 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.05 0.05 0.05
          }
        }
      ]
      name "output 1"
      gradual TRUE
    }
    LED {
      translation 0.2 -0.2 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.05 0.05 0.05
          }
        }
      ]
      name "output 2"
      gradual TRUE
    }
    Shape {
      appearance Pcb {
      }
      geometry DEF plane Plane {
      }
    }
  ]
  boundingObject USE plane
  controller "visualTestController"
}
Ball {
  translation -0.39 0 0.07
  color 0.333333 0 0.498039
  radius 0.05
}
Ball {
  translation 0.32 0 0.07
  name "ball(1)"
  color 0.333333 0 0.498039
  radius 0.05
}
Ball {
  translation -0.01 -0.31 0.07
  name "ball(2)"
  color 0.333333 0 0.498039
  radius 0.05
}
Ball {
  translation 0.1 0.21 0.07
  name "ball(3)"
  color 0.333333 0 0.498039
  radius 0.05
}
Solid {
  translation 0 0.5 0.05
  children [
    Shape {
      appearance MattePaint {
        baseColor 0.666667 0.333333 0
      }
      geometry DEF wallGeom Box {
        size 1 0.01 0.1
      }
    }
  ]
  name "wall"
  boundingObject USE wallGeom
}
Solid {
  translation 0.5 0 0.05
  rotation 0 0 1 -1.5707953071795862
  children [
    Shape {
      appearance MattePaint {
        baseColor 0.666667 0.333333 0
      }
      geometry DEF wallGeom Box {
        size 1 0.01 0.1
      }
    }
  ]
  name "wall(1)"
  boundingObject USE wallGeom
}
Solid {
  translation 0 -0.5 0.05
  children [
    Shape {
      appearance MattePaint {
        baseColor 0.666667 0.333333 0
      }
      geometry DEF wallGeom Box {
        size 1 0.01 0.1
      }
    }
  ]
  name "wall(2)"
  boundingObject USE wallGeom
}
Solid {
  translation -0.5 0 0.05
  rotation 0 0 1 1.5708
  children [
    Shape {
      appearance MattePaint {
        baseColor 0.666667 0.333333 0
      }
      geometry DEF wallGeom Box {
        size 1 0.01 0.1
      }
    }
  ]
  name "wall(3)"
  boundingObject USE wallGeom
}
